auth/basic/middleware.go

```diff
@@ -64,16 +48,19 @@ func parseBasicAuth(auth string) (username, password string, ok bool) {
 	if err != nil {
 		return
 	}
-	cs := string(c)
-	s := strings.IndexByte(cs, ':')
+
+	s := bytes.IndexByte(c, ':')
 	if s < 0 {
 		return
 	}
-	return cs[:s], cs[s+1:], true
+	return c[:s], c[s+1:], true
 }
 
-// AuthMiddleware returns a Basic Authentication middleware for a particular user and password
+// AuthMiddleware returns a Basic Authentication middleware for a particular user and password.
 func AuthMiddleware(requiredUser, requiredPassword, realm string) endpoint.Middleware {
+	requiredUserBytes := sha256.Sum256([]byte(requiredUser))
+	requiredPassBytes := sha256.Sum256([]byte(requiredPassword))
```

Comment:
- [ ] Convert this array to a slice here, rather than down at line 78 each time you use it.
- [ ] Consider using four extra characters to name the variable "requiredPasswordBytes."